src/EnergyPlus/OutputProcessor.hh

```diff
@@ -767,113 +812,113 @@ namespace OutputProcessor {
                               bool writeToSQL);
 
     void WriteReportVariableDictionaryItem(EnergyPlusData &state,
-                                           ReportingFrequency const reportingInterval, // The reporting interval (e.g., hourly, daily)
-                                           StoreType const storeType,
-                                           int const reportID,              // The reporting ID for the data
-                                           int const indexGroupKey,         // The reporting group (e.g., Zone, Plant Loop, etc.)
-                                           std::string const &indexGroup,   // The reporting group (e.g., Zone, Plant Loop, etc.)
+                                           ReportingFrequency reportingInterval, // The reporting interval (e.g., hourly, daily)
+                                           StoreType storeType,
+                                           int reportID,                    // The reporting ID for the data
+                                           int indexGroupKey,               // The reporting group (e.g., Zone, Plant Loop, etc.)
+                                           eTimeStepType const &indexGroup, // The reporting group (e.g., Zone, Plant Loop, etc.)
```

Comment:
Here is an interesting one though, this function, `WriteReportVariableDictionaryItem` took a string `indexGroup`, but internally it was only passing in the `TimeStepType` string argument from SetupOutputVariable.  So I converted it to accept that new enum type.  It required a minor modification to the unit test, and everything is looking good, but I would love for someone to verify that's an alright change.